### ISSUE TITLE
(OUI Next Theme) Observability Notebooks

### ISSUE THREAD

**Issue by KrooshalUX** [opensearch-project/dashboards-observability#769]
_This issue to be transferred to corresponding repository_

I am working on launching new light and dark mode themes provided by [OUI component library](https://oui.opensearch.org/1.2/#/) for a target launch within 2.10. These changes support the vision expressed in [Future Vision for Dashboards](https://github.com/opensearch-project/OpenSearch-Dashboards/issues/4298.)

I have identified the following front end related issues that prevent the theme from appearing complete and potential solutions within this feature:

- [ ] In a few examples within Observability notebooks, text wrapping fails in "code block"

Screenshot taken from [Flights] OpenSearch SQL Quick Start Guide: 

![Screen Shot 2023-07-27 at 11 50 40 AM](https://github.com/opensearch-project/oui/assets/105819041/7a299696-e847-46c1-bbb3-f71bcfd08d61)

**Comment by wbeckler** [comment]
I think there may not be an issue here. The example in the screenshot shows a long string with no spaces. The right way to display it may in fact be to go beyond the text box. For example, the following string is force-wrapped to stay in the text box, but that means that a reader cannot tell how many X's are in the row and it appears like it has spaces when it doesn't. It's not clear to me that it's a better experience to force wrap a long string.

xxxxxxxxxxxxxxxxxxxxxxxxxxxxxxxxxxxxxxxxxxxxxxxxxxxxxxxxxxxxxxxxxxxxxxxxxxxxxxxxxxxxxxxxxxxxxxxxxxxxxxxxxxxxxxxxxxxxxxxxxxxxxxxxxxxxxxxxxxxxxxxxxxxxxxxxxxxxxxxxxxxxxxxxxxxxxxxxxxxxxxxxxxxxxxxxxxxxxxxxxxxxxxxxxxxxxxxxxxxxxxxxxxxxxxxxxxxxxxxx

**Comment by KrooshalUX** [comment]
@wbeckler the other examples wrap and the user is able to consume the entire string. In this example I had to resize my browser beyond my monitors width multiple times to see the end.

**Comment by derek-ho** [comment]
@pjfitzgibbons temporarily assigning to you until we figure out how to divide up these tasks - seems like a lot of OUI compliance issues coming our way

**Comment by pjfitzgibbons** [comment]
> @wbeckler the other examples wrap and the user is able to consume the entire string. In this example I had to resize my browser beyond my monitors width multiple times to see the end. I would expect this to be scrollable if the case is that the wrapping is not desired.

@KrooshalUX could you show us an example of correct wrapping?   
If any correct wrapping requires spaces, then that would be artificial rendering of the users direct input -- so we'd have to think carefully about how to address the situation as found above.

Also, could you tell us where you found the example above?  It might be a degenerate case ?

**Comment by KrooshalUX** [comment]
@peternied The sample came from [Flights] OpenSearch SQL Quick Start Guide. Unfortunately, I no longer have access to this sample notebook. Previous sections in the notebook wrapped "correct" as in they did not break the container as the one in the screenshot does.

**Comment by pjfitzgibbons** [comment]
@KrooshalUX (@pjfitzgibbons )
I'm confused.  Is this sample no longer available?   If this sample is believed-availailable and you no longer have access - could you tell me who POC I can contact?

**Comment by KrooshalUX** [comment]
@pjfitzgibbons its no longer available on the endpoint where we were doing walkthroughs of the new theme, that is correct.

@AMoo-Miki - do you have any additional info here?

I would not say this is critical for 2.10 release.

**Comment by AMoo-Miki** [comment]
I am not sure I follow what @pjfitzgibbons is referring to when he says "wrapping requires spaces"; I believe a simple `word-wrap` and `break-word` is sufficient. [It is used here](https://github.com/opensearch-project/dashboards-observability/blob/d9cb4764f3e031cfbc455ac647a46cf3516505da/public/components/application_analytics/helpers/utils.tsx#L61).

**Comment by pjfitzgibbons** [comment]
Welp.  learn something every day.
We'll work on a fix for this one.

**Comment by pjfitzgibbons** [comment]
Fixed by #974

**Comment by joshuarrrr** [comment]
There's also an OUI utility class available for this:

```
<div className="eui-textBreakWord">
  /* Your content */
</div>
```

https://oui.opensearch.org/1.0/#/utilities/css-utility-classes
